deps/src/singular.cpp

```diff
@@ -206,6 +206,72 @@ JLCXX_MODULE define_julia_module(jlcxx::Module & Singular)
     }
     return old_value;
   });
+  // all of the global setters return the previous value
+  Singular.method("set_option", [](std::string opt, bool value, ring r) {
+    bool old_value = false;
+    ring oldring=currRing;
+    if (r!=NULL) rChangeCurrRing(r);
+    if (false)
+      ;
```

Comment:
?

Reply:
Oh I see this comes the existing `Singular.method("set_option", [](std::string opt, bool value) {` code, I see now